platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/telemetry/PerformanceMeasurementActivity.java

```diff
@@ -0,0 +1,187 @@
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onDestroy() {
+
+    startTimes.clear();
+
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  private void triggerPerformanceEvent(String style, long elapsed) {
+
+    List<Attribute<String>> attributes = new ArrayList<>();
+    attributes.add(
+            new Attribute<>("style_id", style));
+    attributes.add(
+            new Attribute<>("test_perf_event", "true"));
+
+    List<Attribute<Long>> counters = new ArrayList();
+    counters.add(new Attribute<>("elapsed", elapsed));
+
+
+    JsonObject metaData = new JsonObject();
+    metaData.addProperty("os", "android");
+    metaData.addProperty("manufacturer", Build.MANUFACTURER);
+    metaData.addProperty("brand", Build.BRAND);
+    metaData.addProperty("device", Build.MODEL);
+    metaData.addProperty("version", Build.VERSION.RELEASE);
+    metaData.addProperty("abi", Build.CPU_ABI);
+    metaData.addProperty("country", Locale.getDefault().getISO3Country());
+    metaData.addProperty("ram", getRam());
+    metaData.addProperty("screenSize", getWindowSize());
+
+    Gson gson = new Gson();
+
+    Bundle bundle = new Bundle();
+    bundle.putString("attributes", gson.toJson(attributes));
+    bundle.putString("counters", gson.toJson(counters));
+    bundle.putString("metadata", metaData.toString());
+
+    Mapbox.getTelemetry().onPerformanceEvent(bundle);
+  }
+
+  private static String getRam() {
+    ActivityManager actManager =
+            (ActivityManager) Mapbox.getApplicationContext().getSystemService(Context.ACTIVITY_SERVICE);
+    ActivityManager.MemoryInfo memInfo = new ActivityManager.MemoryInfo();
+    actManager.getMemoryInfo(memInfo);
+    return String.valueOf(memInfo.totalMem);
+  }
+
+  private static String getWindowSize() {
+    WindowManager windowManager =
+            (WindowManager) Mapbox.getApplicationContext().getSystemService(Context.WINDOW_SERVICE);
+    Display display = windowManager.getDefaultDisplay();
+    DisplayMetrics metrics = new DisplayMetrics();
+    display.getMetrics(metrics);
+    int width = metrics.widthPixels;
+    int height = metrics.heightPixels;
+
+    return "{" + width + "," + height + "}";
+  }
+
+  private class Attribute<T> {
+    private String name;
+    private T value;
+
+    Attribute(String name, T value) {
+      this.name = name;
+      this.value = value;
+    }
+  }
+}
```

Comment:
above methods are all duplicated across app vs test code. Move to utility instead.